src/circuit/gadget/utilities/cond_swap.rs

```diff
@@ -0,0 +1,319 @@
+use super::{copy, CellValue, UtilitiesInstructions};
+use halo2::{
+    circuit::{Chip, Layouter},
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Permutation, Selector},
+    poly::Rotation,
+};
+use pasta_curves::arithmetic::FieldExt;
+use std::marker::PhantomData;
+
+pub trait CondSwapInstructions<F: FieldExt>: UtilitiesInstructions<F> {
+    /// Variable representing cell with a certain value in the circuit.
+    type Var;
```

Comment:
AFAICT `UtilitiesInstructions` is not used inside `CondSwapInstructions`. Either `CondSwapInstructions::Var` should be removed (using `UtilitiesInstructions::Var` instead), or the `UtilitiesInstructions` bound should be removed.

Reply:
`CondSwapInstructions` now uses `UtilitiesInstructions::Var`.